src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableWithAddMethodConverter.cs

```diff
@@ -79,5 +75,15 @@ protected override bool OnWriteResume(Utf8JsonWriter writer, TCollection value,
 
             return true;
         }
+
+        internal override bool RequiresDynamicMemberAccessors => true;
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091:UnrecognizedReflectionPattern",
+            Justification = "The ctor is marked RequiresUnreferencedCode.")]
```

Comment:
This is dangerous. We now have a ctor that isn't marked with RUC, so this justifciation no longer holds.

Reply:
Would having a separate, reflection-free converter for source-gen usage be a better approach? Luckily we have just 3 converter types that use this pattern, so it wouldn't be a lot of duplication.

Reply:
I think that might be a good idea. I'm a little nervous that while today we don't have an issue here, someone could edit the code later and start calling these methods in different combinations and we wouldn't get warnings because of these invalid suppressions.

Reply:
I opened https://github.com/dotnet/runtime/pull/55846.